samples/bluetooth/peripheral_uart/src/main.c

```diff
@@ -70,104 +77,172 @@ static const struct bt_data sd[] = {
 	BT_DATA_BYTES(BT_DATA_UUID128_ALL, NUS_UUID_SERVICE),
 };
 
-static void uart_cb(struct device *uart)
+static void uart_cb(struct uart_event *evt, void *user_data)
 {
-	static struct uart_data_t *rx;
-
-	uart_irq_update(uart);
-
-	if (uart_irq_rx_ready(uart)) {
-		int data_length;
-
-		if (!rx) {
-			rx = k_malloc(sizeof(*rx));
-			if (rx) {
-				rx->len = 0;
-			} else {
-				/* Disable UART interface, it will be
-				 * enabled again after releasing the buffer.
-				 */
-				uart_irq_rx_disable(uart);
-				rx_disabled = true;
-
-				printk("Not able to allocate UART receive buffer\n");
-				return;
-			}
+	static u8_t *current_buf;
+	static size_t aborted_len;
+	static bool buf_release;
+	struct uart_data_t *buf;
+	static u8_t *aborted_buf;
+
+	switch (evt->type) {
+	case UART_TX_DONE:
+		if ((evt->data.tx.len == 0) ||
+		    (!evt->data.tx.buf)) {
+			return;
 		}
 
-		data_length = uart_fifo_read(uart, &rx->data[rx->len],
-					     UART_BUF_SIZE-rx->len);
-		rx->len += data_length;
-
-		if (rx->len > 0) {
-			/* Send buffer to bluetooth unit if either buffer size
-			 * is reached or the char \n or \r is received, which
-			 * ever comes first
-			 */
-			if ((rx->len == UART_BUF_SIZE) ||
-			   (rx->data[rx->len - 1] == '\n') ||
-			   (rx->data[rx->len - 1] == '\r')) {
-				k_fifo_put(&fifo_uart_rx_data, rx);
-				rx = NULL;
-			}
+		if (aborted_buf) {
+			buf = CONTAINER_OF(aborted_buf, struct uart_data_t,
+					   data);
+			aborted_buf = NULL;
+			aborted_len = 0;
+		} else {
+			buf = CONTAINER_OF(evt->data.tx.buf, struct uart_data_t,
+					   data);
 		}
-	}
 
-	if (uart_irq_tx_ready(uart)) {
-		struct uart_data_t *buf =
-			k_fifo_get(&fifo_uart_tx_data, K_NO_WAIT);
-		u16_t written = 0;
+		k_free(buf);
 
-		/* Nothing in the FIFO, nothing to send */
+		buf = k_fifo_get(&fifo_uart_tx_data, K_NO_WAIT);
 		if (!buf) {
-			uart_irq_tx_disable(uart);
 			return;
 		}
 
-		while (buf->len > written) {
-			written += uart_fifo_fill(uart,
-						  &buf->data[written],
-						  buf->len - written);
+		if (uart_tx(uart, buf->data, buf->len, SYS_FOREVER_MS)) {
+			LOG_WRN("Failed to send data over UART");
```

Comment:
Is it necessary to use Logger functions instead of printk? It would be good to maintain the same style of logging across all samples.

Have you tried using: ``LOG_BACKEND_RTT_FORCE_PRINTK`` configuration?

Reply:
We use `printk` in both uart samples to printk at least initial sample messages to the UART console. That is why a logger is used here.